cpp/src/Ice/ConnectionI.cpp

```diff
@@ -2193,6 +2267,13 @@ Ice::ConnectionI::setState(State state)
             setState(StateClosed, current_exception());
         }
     }
+
+    if (isAtRest())
```

Comment:
Shouldn't we check for `_state == StateActive` here?

Reply:
```cpp
bool isAtRest() const noexcept
{
    return _state == StateActive && _dispatchCount == 0 && _asyncRequests.empty();
}
```